src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs

```diff
@@ -137,7 +138,7 @@ public bool AddToCache (AssemblyDefinition assembly)
 
 			try {
 				assembly  = ReadAssembly (fileName);
-			} catch (Exception e) when (e is FileNotFoundException || e is DirectoryNotFoundException) {
+			} catch (Exception e) when (!(e is SymbolsNotFoundException) && (e is FileNotFoundException || e is DirectoryNotFoundException)) {
```

Comment:
While debugging I found it odd that it returned `null` on `SymbolsNotFoundException` and discovered it inherited from `FileNotFoundException`.

The calling code retries without symbols if we just let it throw `SymbolsNotFoundException` here.

Reply:
If we let the `SymbolsNotFoundException` through here, won't the next `catch` under this one trigger and we'll log a build error? I don't think we want to fail the build if we're missing symbols.

Reply:
It should go here if `SymbolsNotFoundException` is thrown:

https://github.com/xamarin/java.interop/blob/d5afa0af820cb9bb41ad52a113147a4845f09281/src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs#L171-L176

Reply:
Why does it not go here?

https://github.com/xamarin/java.interop/blob/43bc5e5a1c6e205fbaa78c6f985f6bedd412cb9b/src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs#L144-L146

Reply:
I tried the test by itself: https://github.com/jonathanpeppers/java.interop/commit/61a1558e1c326c87991d4e94f0cfc87b6365958a

At one point it was returning `null` here for `SymbolsNotFoundException ` when I had other change, but I don't see that behavior now.

I think we can remove this change actually, thanks.